src/transaction/locator_sr.c

```diff
@@ -13861,4 +13856,16 @@ locator_multi_insert_force (THREAD_ENTRY * thread_p, HFID * hfid, OID * class_oi
 
   return NO_ERROR;
 }
+
+bool
+has_errors_filtered_for_insert (std::vector<int> error_filter_array)
+{
+  if (std::find (error_filter_array.begin(), error_filter_array.end(), ER_BTREE_UNIQUE_FAILED)
```

Comment:
Is it enough to consider only ER_BTREE_UNIQUE_FAILED?

Reply:
I think that for now it should suffice? The problem was that the unique error was filtered using the error file and the `heap_insert` will succeed, while the `btree_insert` will fail. And since the error is filtered, the transaction will eventually commit and the `heap_insert` would not have been rollbacked. I think we can add other errors as well if you have any idea of what could they be.